enhance: Mark query node as read only after suspend

weiliu1031 opened the pull request:
issue: #34985 #35493
after querynode has been suspended, it's not allow to load segment/channel on it, which means the node is read only. to be compatible with resource group design, after query node has been suspend, we remove it from it's original resource group, make it a read only query node in replica. then two things will happens:
1. it's original resource group will be lacking of query nodes, query coord will assign new node to it.
2. querycoord will try to move out all segments/channels after querynode has been suspended

chyezh commented on internal/querycoordv2/meta/resource_manager.go:
```diff
@@ -482,6 +486,55 @@ func (rm *ResourceManager) HandleNodeStopping(node int64) {
+		return nil
+	}
+
+	if err := rm.transferNode(rgName, node); err != nil {
```
In my opinion, call `transfernode` out of recover logic is not a good choice.
HandleNodeResume is same operation with HandleNodeUp,
HandleNodeSuspend is same operation with HandleNodeDown.
Node state is keep by node manager, resource group manager didn't keep any state of node, just put in or pop out.

weiliu1031 commented:
rerun ut

weiliu1031 committed:
enhance: Mark query node as read only after suspend

after querynode has been suspended, it's not allow to load
segment/channel on it, which means the node is read only. to be
compatible with resource group design, after query node has been
suspend, we remove it from it's original resource group, make it a read
only query node in replica. then two things will happens:
1. it's original resource group will be lacking of query nodes, query
   coord will assign new node to it.
2. querycoord will try to move out all segments/channels after querynode
   has been suspended

Signed-off-by: Wei Liu <wei.liu@zilliz.com>

weiliu1031 commented:
/run-cpu-e2e

weiliu1031 commented:
rerun ut

chyezh commented:
\lgtm

weiliu1031 commented:
/hold

weiliu1031 commented:
/unhold

chyezh commented:
/lgtm

congqixia commented:
/approve
